src/components/widgets/Leaderboard.tsx

```diff
@@ -164,11 +164,13 @@ export function Leaderboard(props: Props) {
               class={classes(styles.button.sm, "flex-auto bg-transparent text-white outline-none")}
               placeholder="Search"
               ref={searchInput}
-              onKeyDown={(e) => e.key === "Enter" && setQuery(searchInput?.value)}
+              onKeyDown={(e) =>
+                e.key === "Enter" && setQuery(searchInput?.value.trim() !== "" ? searchInput?.value : undefined)
```

Comment:
I could also simply put `setQuery(searchInput?.value || undefined)`, not sure if we want the trim or not